### PR TITLE
feat(processors): Add processor to split a metric into multiple

### PR DESCRIPTION
**Pull request by powersj**
fixes: #13251

### CODE REVIEW

**Comment by powersj** [comment]
ok - updates made! Let me know if the readme has enough warnings of if you think they should be bigger more bold?

File: plugins/processors/split/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Split Processor Plugin
+
+This plugin splits a metric up into one or more metrics based on a template
+the user provides. The timestamp of the new metric is based on the source
+metric. Templates can overlap, where a field or tag, is used across templates
+and as a result end up in multiple metrics.
+
+If drop original is changed to true, then the plugin can result in dropping
+all metrics when no match is found! Please ensure to test templates before
+putting into production to avoid data loss.
```
Comment (srebhan) [inline plugins/processors/split/README.md:10]:
```suggestion
**NOTE**: If drop original is changed to true, then the plugin can result in
dropping all metrics when no match is found! Please ensure to test
templates before putting into production *and* use metric filtering to
avoid data loss.
```

File: plugins/processors/split/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Some outputs are sensitive to the number of metric series that are produced.
+Multiple series of the same metric name can result in a decrease in performance
+in these outputs.
```
Comment (srebhan) [inline plugins/processors/split/README.md:14]: My concern was the opposite:
```suggestion
Some outputs are sensitive to the number of metric series that are produced.
Multiple metrics of the same series (i.e. identical name, tag key-values and
field name) with the same timestamp might result in squashing those points
to the latest metric produced.
```

**Review by srebhan — COMMENTED** [review]
Two small comments regarding the README warnings...

**Pull request merged**